segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java

```diff
@@ -989,11 +990,11 @@ <R> CompletableFuture<R> executeParallel(Callable<CompletableFuture<R>> operatio
         }, this.executor);
     }
 
-    static <T extends AutoCloseable, R> CompletableFuture<R> tryWith(T closeable, Function<T, CompletableFuture<R>> function, Executor executor) {
-        return function.apply(closeable)
+    static <R> CompletableFuture<R> tryWith(MetadataTransaction txn, Function<MetadataTransaction, CompletableFuture<R>> function, Executor executor) {
```

Comment:
This change is irrelevant and not needed in this PR.

Reply:
I just tested reverting this line and it gave me multiple compile errors.